lib/puppet/pops/evaluator/closure.rb

```diff
@@ -219,24 +219,25 @@ def enclosing_scope
   def call_with_scope(scope, args)
     variable_bindings = combine_values_with_parameters(scope, args)
 
-    tc = Types::TypeCalculator.singleton
-    final_args = tc.infer_set(parameters.reduce([]) do |tmp_args, param|
+    final_args = parameters.reduce([]) do |tmp_args, param|
       if param.captures_rest
         tmp_args.concat(variable_bindings[param.name])
       else
         tmp_args << variable_bindings[param.name]
       end
-    end)
+    end
 
-    if type.callable?(final_args)
+    if type.callable_with?(final_args, nil)
```

Comment:
This works because it just so happens Closure's are always created with a `nil` block type: https://github.com/puppetlabs/puppet/blob/50d3184fba4f43957a371e57ad09cadaf1b7a268/lib/puppet/pops/evaluator/closure.rb#L312

But to be future proof, we should pass the `block_type` when checking if the `type` is callable.